lib/rspec/rails/example.rb

```diff
@@ -1,13 +1,16 @@
 require 'rspec/rails/example/rails_example_group'
-require 'rspec/rails/example/controller_example_group'
-require 'rspec/rails/example/request_example_group'
-require 'rspec/rails/example/helper_example_group'
-require 'rspec/rails/example/view_example_group'
-require 'rspec/rails/example/mailer_example_group'
-require 'rspec/rails/example/routing_example_group'
 require 'rspec/rails/example/model_example_group'
-require 'rspec/rails/example/job_example_group'
 require 'rspec/rails/example/feature_example_group'
-require 'rspec/rails/example/system_example_group'
 require 'rspec/rails/example/channel_example_group'
 require 'rspec/rails/example/mailbox_example_group'
+require 'rspec/rails/example/mailer_example_group' if defined?(ActionMailer)
+require 'rspec/rails/example/job_example_group' if defined?(ActiveJob)
+require 'rspec/rails/example/view_example_group' if defined?(ActionView)
+
+if defined?(ActionPack)
```

Comment:
If we go ahead with this, this is not my preferred way to handle this, the constants must always exist because extensions may expect them to exist. They are part of our public API and semver means we must maintain this.

Reply:
Agree, valid poin.